compiler/parser.jou

```diff
@@ -212,11 +213,115 @@ def decorate(stmt: AstStatement*, decor: byte*, location: Location) -> None:
         fail(location, msg)
 
 
+enum WhatAreWeParsing:
+    TopLevel  # must be first, so that this is the default (zero)
+    ClassBody
+    FunctionBody
+    MethodBody
+
+
+# Errors if e.g. trying to define a class inside a function (other way around it would become a method)
+def check_statement_not_misplaced(
```

Comment:
In many cases it is possible to emit error earlier, before this function can be called. For example, when we start parsing a class, we could check if it's in a function or method or other class. This is better, because if class contains errors, we tell user not to put class inside class or whatever, so they stop trying to do it early.